src/simple_cache_client.rs

```diff
@@ -497,6 +501,181 @@ impl SimpleCacheClient {
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use uuid::Uuid;
+    /// use std::num::NonZeroU64;
+    /// # tokio_test::block_on(async {
+    ///     use std::env;
+    ///     use momento::{
+    ///         response::cache_dictionary_get_response::MomentoDictionaryGetStatus,
+    ///         simple_cache_client::SimpleCacheClientBuilder,
+    ///     };
+    ///     let auth_token = env::var("TEST_AUTH_TOKEN").expect("TEST_AUTH_TOKEN must be set");
+    ///     let cache_name = Uuid::new_v4().to_string();
+    ///     let mut momento = SimpleCacheClientBuilder::new(auth_token, NonZeroU64::new(30).unwrap())
+    ///         .expect("could not create a client")
+    ///         .build();
+    ///     momento.create_cache(&cache_name).await;
+    ///
+    ///     let dictionary_name = Uuid::new_v4().to_string();
+    ///
+    ///     let resp = momento.dictionary_get(&cache_name, &dictionary_name, vec![
+    ///         "key1".to_string(),
+    ///         "key2".to_string(),
+    ///     ]).await.unwrap();
+    ///
+    ///     match resp.result {
+    ///         MomentoDictionaryGetStatus::FOUND => println!("dictionary found!"),
+    ///         MomentoDictionaryGetStatus::MISSING => println!("dictionary missing!"),
+    ///         _ => println!("error occurred")
+    ///     };
+    ///
+    ///
+    ///     if let Some(dictionary) = resp.dictionary {
+    ///         println!("dictionary entries:");
+    ///         for (key, value) in dictionary.iter() {
+    ///             println!("{:?} => {:?}", key, value);
+    ///         }
+    ///     }
+    ///
+    ///     momento.delete_cache(&cache_name).await;
+    /// # })
+    /// ```
+    pub async fn dictionary_get<K: MomentoRequest>(
+        &mut self,
+        cache_name: &str,
+        dictionary: &str,
+        fields: Vec<K>,
+    ) -> Result<MomentoDictionaryGetResponse, MomentoError> {
+        utils::is_cache_name_valid(cache_name)?;
+        let mut fields = fields;
+
+        let mut fields: Vec<Vec<u8>> = fields.drain(..).map(|f| f.into_bytes()).collect();
+
+        let mut request = tonic::Request::new(DictionaryGetRequest {
+            dictionary_name: dictionary.into_bytes(),
+            fields: fields.clone(),
+        });
+        request_meta_data(&mut request, cache_name)?;
+        let response = self.data_client.dictionary_get(request).await?.into_inner();
+        match response.dictionary {
+            Some(Dictionary::Found(mut response)) => {
+                let mut dictionary = HashMap::new();
+
+                // map the dictionary response parts to get responses
+                for (field, item) in fields.drain(..).zip(response.items.drain(..)) {
+                    match item.result() {
+                        ECacheResult::Hit => {
+                            dictionary.insert(field, item.cache_body);
+                        }
+                        ECacheResult::Miss => {}
+                        _ => todo!(),
```

Comment:
@malandis what was the result we wanted here?

Reply:
Same thing here @cprice404

Reply:
I think this is the relevant code from .NET:

https://github.com/momentohq/client-sdk-dotnet-incubating/blob/main/src/Momento.Sdk.Incubating/Internal/ScsDataClient.cs#L315

Reply:
Thanks Chris! This has been addressed